server/dashboards/preconfigured/cosmosdb-handoff.ts

```diff
@@ -0,0 +1,234 @@
+      props: {
+        cols: [
+          { header: 'Timestamp', field: 'timestamp', type: 'time', format: 'MMM-DD HH:mm:ss' },
+          { header: 'Last Message', field: 'lastMessage' },
+          { header: 'Username', field: 'username' },
+          { header: 'Status', field: 'icon', type: 'icon' },
+          { type: 'button', value: 'chat', click: 'openTranscriptsDialog' }
+        ]
+      },
+      actions: {
+        openTranscriptsDialog: {
+          action: 'dialog:transcripts',
+          params: { title: 'args:username', conversationId: 'args:conversationId', queryspan: 'timespan:queryTimespan' }
+        }
+      }
+    }
+  ],
+  dialogs: [
+    {
+      id: 'transcripts',
+      width: '60%',
+      params: ['title', 'conversationId', 'queryspan'],
+      dataSources: [
+        {
+          id: 'transcripts-data',
+          type: 'CosmosDB/Query',
+          dependencies: {
+            conversationId: 'dialog_transcripts:conversationId',
+            queryTimespan: 'dialog_transcripts:queryspan',
+            secret: 'connection:bot-framework.directLine'
+          },
+          params: {
+            databaseId: 'admin',
+            collectionId: 'conversations',
+            query: ({ conversationId }) => `SELECT * FROM conversations c 
+              WHERE (c.customer.conversation['$id'] = '${conversationId}')`,
+            parameters: []
+          },
+          calculated: (result, dependencies) => {
+            if (!result.Documents) {
+              return null;
+            }
+
+            let values = [];
+            let customer = null;
+            let body = {};
+            let headers = {};
+            let disabled = false;
+            const { secret } = dependencies;
+
+            if (result.Documents.length === 1) {
+              values = result.Documents[0].transcript || [];
+              customer = result.Documents[0].customer;
+              disabled = result.Documents[0].state !== 0 ? true : false;
+              body = {
+                'conversationId': customer.conversation.id,
+              };
+              headers = {
+                'Authorization': `Bearer ${secret}`
+              };
+            }
+            return { 'values': values, 'customer': customer, 'headers': headers, 'body': body, 'disabled': disabled };
+          }
+        }
+      ],
+      elements: [
+        {
+          id: 'transcripts-button',
+          type: 'RequestButton',
+          title: 'Transfer to Agent',
+          size: { w: 2, h: 1 },
+          location: { x: 0, y: 0 },
+          dependencies: {
+            body: 'transcripts-data:body',
+            headers: 'transcripts-data:headers',
+            disabled: 'transcripts-data:disabled',
+            conversationsEndpoint: 'connection:bot-framework.conversationsEndpoint'
+          },
+          props: {
+            url: ({ conversationsEndpoint }) => `${conversationsEndpoint}`,
+            method: 'POST',
+            once: true,
```

Comment:
What I meant is about `once` - maybe this should be renames to `disableAfterClick`